lib/rugged_adapter/git_layer_rugged.rb

```diff
@@ -188,13 +188,29 @@ def cat_file(options, sha)
         @repo.lookup(sha).read_raw
       end
 
-      def apply_patch(head_sha = 'HEAD', patch=nil)
-        false # Rewrite gollum-lib's revert so that it doesn't require a direct equivalent of Grit's apply_patch
-      end
-
-      def revert(path, sha1, sha2, ref)
-        # FIXME: See https://github.com/gollum/grit_adapter/pull/14
-        fail NotImplementedError
+      def revert(path, sha1, sha2)
```

Comment:
You are checking if `path` exists in three different places. I'd like to see this refactored to one. 

Also, what is the case for revert without `path` set?

Reply:
Will do.

The case is that of reverting an entire commit, rather than changes to just one file.